frontend/app/services/EmailService.scala

```diff
@@ -14,18 +19,27 @@ trait EmailService extends LazyLogging {
   val feedbackAddress: String
 
   val client = {
-    val c= new AmazonSimpleEmailServiceClient()
+    val c = new AmazonSimpleEmailServiceClient()
     c.setRegion(Region.getRegion(Regions.EU_WEST_1))
     c
   }
 
+  private val digest = MessageDigest.getInstance("MD5")
+
+  def md5(input: String): String = {
+    val hf = Hashing.md5()
+    util.Arrays.toString(hf.newHasher().putBytes(input.getBytes("UTF-8")).hash().asBytes())
```

Comment:
You can write this as `Hashing.md5().hashString("your input", StandardCharsets.UTF_8).toString` <- note the `toString` method returns a nice hex string...

...or alternatively you could construct the blacklisted email as a `HashCode` object using `HashCode.fromString()`, and pass that around as a more strongly typed alternative to strings..!